### ISSUE TITLE
Is it an error that output is further minifiable?

### ISSUE THREAD

**Issue by krk** [google/closure-compiler#1946]
Does running closure in ADVANCED mode on its own output minifies the output further?

I have a case when it does, trying to figure out if this is a bug in my custom passes or an expected behaviour.

Example in lengths, as without the pass source code, it would be meaningless: 
10000 bytes input -> 1000 bytes output.
Same 1000 bytes output -> 100 bytes output.

When my custom pass replaces nodes, it marks the compiler with `compiler.reportCodeChange();`, still it can optimize the final output of closure.

In other words, how can I make sure that several custom passes has run "enough" times? - They are added as `isOneTimePass:false` to `PassFactory`

**Comment by dimvar** [comment]
(In the future, please ask questions at https://groups.google.com/forum/#!forum/closure-compiler-discuss. The github issue tracker is for bug reports.)


Generally, we don't provide any guarantees that the compiler output will be in a state where it can be compiled without errors. It may happen sometimes, but don't count on it.

If you recompile some output and get a smaller result, it may be because of some miscompilation that removed a bunch of code that should have stayed live, eg, if you forget to include the externs in the second compilation, or if the output is obscure enough that the compiler can't find uses of some function, and deletes it.

It can also be the case that you recompile some output, you get a smaller result, and the result is correct. This can happen if at some point late in the first compilation, a pass exposes some optimization opportunities for some other pass that has already run, and won't be run again.

> how can I make sure that several custom passes has run "enough" times?

There isn't a certain way to know that. You have to use heuristics and experimentation.
